cmd/ulxly/ulxly.go

```diff
@@ -126,6 +182,294 @@ var DepositsCmd = &cobra.Command{
+			return err
+		}
+
+		privateKey, err := crypto.HexToECDSA(*ulxlyInputArgs.PrivateKey)
+		if err != nil {
+			log.Error().Err(err).Msg("Unable to retrieve private key")
+		}
+
+		publicKey := privateKey.Public()
+		publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+		if !ok {
+			log.Error().Msg("Error casting public key to ECDSA")
+		}
+
+		fromAddress := crypto.PubkeyToAddress(*publicKeyECDSA)
+
+		value := big.NewInt(*ulxlyInputArgs.Value)
+		gasLimit := *ulxlyInputArgs.GasLimit // Deposit transactions seem to use >300,000 units of gas to be safe.
+		gasPrice, err := client.SuggestGasPrice(ctx)
+		if err != nil {
+			log.Error().Err(err).Msg("Cannot get suggested gas price")
+		}
+		// gasTipCap, err := client.SuggestGasTipCap(ctx)
+		// if err != nil {
+		// 	log.Error().Err(err).Msg("Cannot get suggested gas tip cap")
+		// }
+
+		toAddress := common.HexToAddress(*ulxlyInputArgs.DestinationAddress)
+		tokenAddress := common.HexToAddress(*ulxlyInputArgs.TokenAddress)
+		metaBytes := common.Hex2Bytes(*ulxlyInputArgs.MetaBytes)
+
+		chainID, err := client.NetworkID(ctx)
+		if err != nil {
+			log.Error().Err(err).Msg("Cannot get chain ID")
+		}
+
+		signer := types.LatestSignerForChainID(chainID)
+
+		var tops *bind.TransactOpts
+		if tokenAddress == common.HexToAddress("0x0000000000000000000000000000000000000000") {
+			tops = &bind.TransactOpts{
+				Signer: func(address common.Address, transaction *types.Transaction) (*types.Transaction, error) {
+					return types.SignTx(transaction, signer, privateKey)
+				},
+				Value:     value,
+				From:      fromAddress,
+				Context:   ctx,
+				GasLimit:  gasLimit,
+				GasPrice:  gasPrice,
+				GasFeeCap: nil,
+				GasTipCap: nil,
+			}
+		} else {
+			tops = &bind.TransactOpts{
+				Signer: func(address common.Address, transaction *types.Transaction) (*types.Transaction, error) {
+					return types.SignTx(transaction, signer, privateKey)
+				},
+				From:      fromAddress,
+				Context:   ctx,
+				GasLimit:  gasLimit,
+				GasPrice:  gasPrice,
+				GasFeeCap: nil,
+				GasTipCap: nil,
+			}
+		}
+
+		bridgeAssetTxn, err := bridgeV2.BridgeAsset(tops, *ulxlyInputArgs.DestinationNetwork, toAddress, value, tokenAddress, *ulxlyInputArgs.IsForced, metaBytes)
+		if err != nil {
+			return err
+		}
+
+		// Wait for the transaction to be mined
+		for {
+			r, err := client.TransactionReceipt(ctx, bridgeAssetTxn.Hash())
+			if err != nil {
+				if err.Error() != "not found" {
+					log.Error().Err(err)
+				}
+				time.Sleep(1 * time.Second)
+				continue
+			}
+			if r.Status != 0 {
+				fmt.Printf("Deposit Transaction Successful: %s\n", r.TxHash)
+				break
+			} else if r.Status == 0 {
+				fmt.Printf("Deposit Transaction Failed: %s\n", r.TxHash)
+				fmt.Printf("Try increasing the gas limit:\n")
+				fmt.Printf("Current gas limit: %d\n", gasLimit)
+				fmt.Printf("Cumulative gas used for transaction: %d\n", r.CumulativeGasUsed)
+				break
+			}
+			time.Sleep(1 * time.Second)
+		}
```

Comment:
Since this loop might take forever, it would be great to add a status message (maybe each 10s) that says `Waiting for deposit transaction to be mined...`.